src/System.Drawing.Common/tests/ColorTranslatorTests.cs

```diff
@@ -0,0 +1,256 @@
+        {
+            Assert.Equal(expected, ColorTranslator.FromHtml(htmlColor));
+        }
+
+        [Theory]
+        [InlineData("'", typeof(Exception))]
+        [InlineData("'\"", typeof(Exception))]
+        [InlineData("\"'", typeof(Exception))]
+        [InlineData("#", typeof(Exception))]
+        [InlineData("#G12", typeof(FormatException))]
+        [InlineData("#G12345", typeof(FormatException))]
+        [InlineData("  #G12  ", typeof(Exception))]
+        [InlineData("  #G12345  ", typeof(Exception))]
+        [InlineData("#FFFFFFFFF", typeof(Exception))]
+        [InlineData("0x", typeof(Exception))]
+        [InlineData("0xFFFFFFFFF", typeof(Exception))]
+        [InlineData("0xG12", typeof(Exception))]
+        [InlineData("&h", typeof(Exception))]
+        [InlineData("&hG12", typeof(Exception))]
+        [InlineData("1,2", typeof(ArgumentException))]
+        [InlineData("1,2,3,4,5", typeof(ArgumentException))]
+        [InlineData("-1,2,3", typeof(ArgumentException))]
+        [InlineData("256,2,3", typeof(ArgumentException))]
+        [InlineData("1,-1,3", typeof(ArgumentException))]
+        [InlineData("1,256,3", typeof(ArgumentException))]
+        [InlineData("1,2,-1", typeof(ArgumentException))]
+        [InlineData("1,2,256", typeof(ArgumentException))]
+        public void FromHtml_Invalid_Throws(string htmlColor, Type exception)
+        {
+            Assert.Throws(exception, () => ColorTranslator.FromHtml(htmlColor));
+        }
+
+        public static IEnumerable<object[]> ToHtml_TestData()
+        {
+            yield return new object[] { Color.Empty, "" };
+
+            foreach ((string htmlColor, Color color) in HtmlColors_TestData())
+            {
+                yield return new object[] { color, htmlColor };
+            }
+
+            // SystemColors.ControlLight don't roundtrip.
+            yield return new object[] { SystemColors.ControlLight, "buttonface" };
+            yield return new object[] { SystemColors.GradientActiveCaption, "activecaption" };
+            yield return new object[] { SystemColors.HotTrack, "highlight" };
+            yield return new object[] { SystemColors.MenuHighlight, "highlighttext" };
+            yield return new object[] { SystemColors.GradientInactiveCaption, "inactivecaption" };
+            yield return new object[] { SystemColors.MenuBar, "menu" };
+            yield return new object[] { SystemColors.ButtonShadow, "" };
+        }
+
+        [Theory]
+        [MemberData(nameof(ToHtml_TestData))]
+        public void ToHtml_Color_ReturnsExpected(Color color, string expected)
+        {
+            Assert.Equal(expected, ColorTranslator.ToHtml(color));
+        }
+
+        [Fact]
+        public void Ctor_Default_Success()
+        {
+            ConstructorInfo constructor = typeof(ColorTranslator).GetConstructor(BindingFlags.NonPublic | BindingFlags.Instance, null, new Type[0], null);
```

Comment:
ColorTranslator has an empty private default constructor that can't be reached without this.
Maybe I should just delete that private ctor in the product code, but I wasn't sure about compat concerns so wanted to test for it